plugins/aea-ledger-ethereum/aea_ledger_ethereum/ethereum.py

```diff
@@ -1246,57 +1246,27 @@ def build_transaction(  # pylint: disable=too-many-arguments
         """
         method = getattr(contract_instance.functions, method_name)
         tx = method(**method_args)
-        tx = self._build_transaction(
-            tx,
-            tx_args["sender_address"],
-            tx_args["eth_value"],
-            tx_args["gas"],
-            tx_args["gas_price"],
-            tx_args["max_fee_per_gas"],
-            tx_args["max_priority_fee_per_gas"],
-        )
```

Comment:
I had to change this as obviously some of these fields might be missing and we have not yet checked for them being present at this stage.